lib/Optimizer/Transforms/DecompositionPatterns.cpp

```diff
@@ -137,6 +138,82 @@ struct HToPhasedRx : public OpRewritePattern<quake::HOp> {
   }
 };
 
+// quake.exp_pauli(theta) target pauliWord
+// ───────────────────────────────────
+// Basis change operations, cnots, rz(theta), adjoint basis change
+struct ExpPauliDecomposition : public OpRewritePattern<quake::ExpPauliOp> {
+  using OpRewritePattern::OpRewritePattern;
+
+  void initialize() { setDebugName("ExpPauliDecomposition"); }
+
+  LogicalResult matchAndRewrite(quake::ExpPauliOp expPauliOp,
+                                PatternRewriter &rewriter) const override {
+    auto loc = expPauliOp.getLoc();
+    auto qubits = expPauliOp.getQubits();
+    auto theta = expPauliOp.getParameter();
+    auto pauliWord = expPauliOp.getPauli();
+
+    // Assert that we have a constant known pauli word
+    auto defOp = pauliWord.getDefiningOp<cudaq::cc::CreateStringLiteralOp>();
+    if (!defOp)
+      return failure();
+
+    std::vector<Value> qubitSupport;
+    std::vector<std::function<void(bool)>> basisChange;
+    StringRef pauliWordStr = defOp.getStringLiteral();
+    for (std::size_t i = 0; i < pauliWordStr.size(); i++) {
+      Value index = rewriter.create<arith::ConstantIntOp>(loc, i, 64);
+      Value qubitI = rewriter.create<quake::ExtractRefOp>(loc, qubits, index);
+      if (pauliWordStr[i] != 'I')
+        qubitSupport.push_back(qubitI);
+
+      if (pauliWordStr[i] == 'Y') {
+        APFloat d(M_PI_2);
+        Value param = rewriter.create<arith::ConstantFloatOp>(
+            loc, d, rewriter.getF64Type());
+        rewriter.create<quake::RxOp>(loc, ValueRange{param}, ValueRange{},
+                                     ValueRange{qubitI});
+      } else if (pauliWordStr[i] == 'X')
+        rewriter.create<quake::HOp>(loc, ValueRange{qubitI});
+    }
+
+    std::vector<std::pair<Value, Value>> toReverse;
+    for (std::size_t i = 0; i < qubitSupport.size() - 1; i++) {
+      rewriter.create<quake::XOp>(loc, ValueRange{},
```

Comment:
```suggestion
      rewriter.create<quake::XOp>(loc, 
```
There is supposed to be a builder that lets us omit parameters here. (QuakeOps.td:714)